src/display/annotation_layer.js

```diff
@@ -315,6 +335,16 @@ var TextAnnotationElement = (function TextAnnotationElementClosure() {
 
   Util.inherit(TextAnnotationElement, AnnotationElement, {
     /**
+     * Check does this annotation need HTML element.
+     *
+     * @public
+     * @memberof TextAnnotationElement
+     */
+    hasHtml: function TextAnnotationElement_hasHtml() {
+      return this.data.hasPopup || this.data.title || this.data.contents;
```

Comment:
Where is this logic? I can only see https://github.com/mozilla/pdf.js/pull/6856/files#diff-18ca06e2bd5be4c9132b51f78388b8f0L661 and later check `title`/`contents` of the TextAnnotation https://github.com/mozilla/pdf.js/pull/6856/files#diff-18ca06e2bd5be4c9132b51f78388b8f0L683